fendermint/vm/snapshot/src/car/mod.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! CAR file chunking utilities
+//!
+//! See https://ipld.io/specs/transport/car/carv1/
+
+use anyhow::{self, Context as AnyhowContext};
+use futures::{future, StreamExt};
+use std::path::Path;
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+use fvm_ipld_car::{CarHeader, CarReader};
+
+use self::{chunker::ChunkWriter, streamer::BlockStreamer};
+
+mod chunker;
+mod streamer;
+
+/// Take an existing CAR file and split it up into an output directory by creating
+/// files with a limited size for each file.
+///
+/// The first (0th) file will be just the header, with the rest containing the "content" blocks.
+///
+/// Returns the number of chunks created.
+pub async fn split<F>(
+    input_file: &Path,
+    output_dir: &Path,
+    max_size: usize,
+    file_name: F,
+) -> anyhow::Result<usize>
+where
+    F: Fn(usize) -> String + Send + Sync + 'static,
+{
+    let file = tokio::fs::File::open(input_file)
+        .await
+        .with_context(|| format!("failed to open CAR file: {}", input_file.to_string_lossy()))?;
+
+    let reader: CarReader<_> = CarReader::new_unchecked(file.compat())
+        .await
+        .context("failed to open CAR reader")?;
+
+    // Create a Writer that opens new files when the maximum is reached.
+    let mut writer = ChunkWriter::new(output_dir.into(), max_size, file_name);
+
+    let header = CarHeader::new(reader.header.roots.clone(), reader.header.version);
```

Comment:
Does the cid roots of the header currently point to the cid of the different parts? I don't think this is the case, right? We are just taking the whole CAR as a single DAG and then splitting it into parts.

Reply:
No, it doesn't. It is as you describe, we just split the body of the CAR with no regard to forming sub-dags or anything. The only criteria is the size, and that the content should stay with its CID, so we can validate parts as we get them:
1. Check that the CID corresponds to the content
2. Check that the CID is something that was pointed at by an earlier `Link`.